### ISSUE TITLE
--merge-pr refuses to merge easyconfig PRs because test suite is "pending"

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-framework#3405]
Example:

```
$ eb --merge-pr 11090
== temporary log file in case of crash /tmp/eb-2lDSgs/easybuild-DgLSzN.log

easybuilders/easybuild-easyconfigs PR #11090 was submitted by edmondac, you are using GitHub account 'boegel'

Checking eligibility of easybuilders/easybuild-easyconfigs PR #11090 for merging...
* targets develop branch: OK
* test suite passes: pending... => not eligible for merging!
...
```

This is not correct, because the most recent commit in https://github.com/easybuilders/easybuild-easyconfigs/pull/11090 got a green check mark (see https://github.com/easybuilders/easybuild-easyconfigs/commit/0f37d99bf28dc835ee0046ef8cbce615f96b8c9f).

**Comment by boegel** [comment]
I've been in touch with GitHub Support over this...

It turns out that the "commit status" (cfr. https://docs.github.com/en/rest/reference/repos#list-commit-statuses-for-a-reference and https://docs.github.com/en/rest/reference/repos#get-the-combined-status-for-a-specific-reference) only applies to the tests being by Travis CI, because GitHub Actions doesn't set the commit status but performs "check runs" (cfr. https://docs.github.com/en/rest/reference/checks#list-check-runs-for-a-git-reference and https://docs.github.com/en/rest/reference/checks#list-check-suites-for-a-git-reference).

By no longer testing easyconfig PRs with Travis (see https://github.com/easybuilders/easybuild-easyconfigs/pull/11055), we have nothing that is setting the commit status anymore, so it remains stuck in "`pending`"  forever.

To fix this, we have to update the logic used by `--merge-pr` to determine whether a PR has been given the green light to take into account *both* the "combined commit status" and the "check suites" results, and take into account that the "combined commit status" should be ignored if its "pending" and nothing is running to set the commit status (see for example https://api.github.com/repos/easybuilders/easybuild-easyconfigs/commits/06f73629a61d33485883ebb7079feed5f5d992aa/status, where `state` is `pending and `total_count` is `0`).

**Comment by boegel** [comment]
Problem should be fixed with #3406...
